api/v1alpha1/fenceagentsremediationtemplate_validator.go

```diff
@@ -0,0 +1,316 @@
+
+	return allWarnings, utilErrors.NewAggregate(allErrors)
+}
+
+// ValidateUpdate implements webhook.CustomValidator so a webhook will be registered for the type
+func (v *customValidator) ValidateUpdate(ctx context.Context, old runtime.Object, new runtime.Object) (admission.Warnings, error) {
+	r := new.(*FenceAgentsRemediationTemplate)
+	webhookTemplateValidatorLog.Info("validate update", "name", r.Name)
+
+	var allErrors []error
+	var allWarnings []string
+
+	// First, run the existing FAR validation logic
+	validateWarnings, validateFarErr := validateFAR(&r.Spec.Template.Spec)
+	if validateFarErr != nil {
+		allErrors = append(allErrors, validateFarErr)
+	}
+	// Add validateFAR warnings
+	allWarnings = append(allWarnings, validateWarnings...)
+
+	// Perform enhanced parameter validation with secret collection
+	paramWarnings, validateParamErr := v.validateFenceAgentTemplate(ctx, r)
+	if validateParamErr != nil {
+		allErrors = append(allErrors, validateParamErr)
+	}
+	// Add parameter validation warnings
+	allWarnings = append(allWarnings, paramWarnings...)
+
+	return allWarnings, utilErrors.NewAggregate(allErrors)
+}
+
+// ValidateDelete implements webhook.CustomValidator so a webhook will be registered for the type
+func (v *customValidator) ValidateDelete(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	r := obj.(*FenceAgentsRemediationTemplate)
+	webhookTemplateValidatorLog.Info("validate delete", "name", r.Name)
+	return nil, nil
+}
+
+// validateFenceAgentTemplate validates fence agent parameters for templates
+// by creating temporary FAR CRs and using BuildFenceAgentParams + validateParametersWithStatus
+func (v *customValidator) validateFenceAgentTemplate(ctx context.Context, r *FenceAgentsRemediationTemplate) ([]string, error) {
+	var warnings []string
+	spec := &r.Spec.Template.Spec
+
+	// Check if template has any parameters at all
+	hasSharedParams := len(spec.SharedParameters) > 0
+	hasNodeParams := len(spec.NodeParameters) > 0
+	hasSecrets := spec.SharedSecretName != nil || spec.NodeSecretNames != nil
+
+	// If template has no parameters or secrets, skip parameter validation
+	// Templates are allowed to be empty - parameters can be added later
+	//TODO mshitrit should we allow this ?
+	if !hasSharedParams && !hasNodeParams && !hasSecrets {
+		webhookTemplateValidatorLog.Info("validateFenceAgentTemplate return no params")
+		return warnings, nil
+	}
+
+	// Collect all unique node names from NodeParameters and NodeSecretNames
+	nodeNames := getNodeNamesFromSpec(spec)
+
+	skipStatusValidation := false
+	// If no node-specific parameters, validate with shared parameters only, use a dummy placeholder for node name
+	if len(nodeNames) == 0 {
+		webhookTemplateValidatorLog.Info("validateFenceAgentTemplate no nodes found")
+		nodeNames["temp-validation"] = true
+		// Status validation will NOT occur for shared params with a node template (because we want to avoid getting all the nodes from the API server)
```

Comment:
If we have no node-specific parameters, can we consider picking 1 Node only and perform validation on that one?

Reply:
If we do that we'll need to get it from the API server instead of the template.
I'm not sure how I feel about it.
It seems like it exceeds template validation and I also worry about picking random node which might never intended to be remediated.

Reply:
If there is a selection of nodes not indented for remediation, there wouldn't be a list of nodes intended to remediate in the template?